src/Discord.Net.WebSocket/BaseSocketClient.Events.cs

```diff
@@ -39,6 +40,13 @@ public partial class BaseSocketClient
             remove { _messageDeletedEvent.Remove(value); }
         }
         internal readonly AsyncEvent<Func<Cacheable<IMessage, ulong>, ISocketMessageChannel, Task>> _messageDeletedEvent = new AsyncEvent<Func<Cacheable<IMessage, ulong>, ISocketMessageChannel, Task>>();
+        /// <summary> Fired when multiple messages bulk deleted. </summary>
+        public event Func<List<Cacheable<IMessage, ulong>>, ISocketMessageChannel, Task> MessageBulkDeleted {
```

Comment:
Change the collection type to be an `IReadOnlyCollection<Cacheable<IMessage, ulong>>`